autoinstrumentation/java/Dockerfile

```diff
@@ -8,6 +8,10 @@ FROM busybox
 
 ARG version
 
-ADD https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/download/v$version/opentelemetry-javaagent.jar /javaagent.jar
+RUN addgroup otel -g 1000 && adduser -u 1000 -D -G otel otel
+
+USER 1000:1000
```

Comment:
I might be wrong, but ive something in mind - Was there an issue with something like this on OpenShift?

ping @iblancasa @pavolloffay

Reply:
@frzifus will this Openshift thing block the merge?

Reply:
I took sometime to see how that works in Openshift and yeah - they are dealing with this on their side not to run it with root and docker image with explicit UID is introducing issues for them.